helix-core/src/main/java/org/apache/helix/messaging/handling/HelixTaskExecutor.java

```diff
@@ -701,7 +701,7 @@ private void syncSessionToController(HelixManager manager) {
       }
     }
 
-    List<Message> newMessages = accessor.getProperty(keys);
+    List<Message> newMessages = accessor.getProperty(keys, true);
```

Comment:
This is not a good idea. If we throws exception here, then there is no event to trigger message read. Then it could only relies on @mgao0 's periodical message refresh. But if user does not set this up?

Do not throw exception at least can read some message back and process them. Also when ST done with these partial messages, delete message operation can trigger another round of read. I know relies on this logic is not robust. But at least, it is not worse scenario than throw exception here.

Reply:
Good point. I will change it back and add this to the comment.
Please help to check the other changes. My goal is to make all calls strictly checked unless we have a good reason.